### ISSUE TITLE
单卡训练没有问题，多卡情况会OOM

### ISSUE THREAD

**Issue by Tttizi** [ByteDance-Seed/VeOmni#159]
单卡训练的时候是可以正常训练的，相同参数的情况下多卡出现了OOM，单卡训练时有20GB的显存是盈余的，多卡会炸掉。有什么推荐的设置吗？

**Comment by Luosuu** [comment]
Hi @Tttizi ,

could you please share the job yaml you are running?

**Comment by Tttizi** [comment]
@Luosuu 
model:
  config_path: XXX
  tokenizer_path: XXX
  visual_weights_path: XXX
  attn_implementation: flash_attention_2
  torch_dtype: bfloat16

data:
  data_type: conversation
  dataloader_type: native
  datasets_type: iterable
  chat_template: llava
  text_keys: conversations
  max_seq_len: 4096
  train_path: XXX
  image_folder: XXX
  scale_resolution: 512
  upscale_rate: 1
  drop_last: true

train:
  lr: 2e-4
  weight_decay: 0.
  optimizer: adamw

  max_steps: 2180
  micro_batch_size: 1
  global_batch_size: 8
  num_train_epochs: 1

  lr_decay_style: cosine
  lr_decay_ratio: 1.0
  lr_warmup_ratio: 0.03
  lr_start: 0.
  max_grad_norm: 1.0

  trainable_modules: ["projector","embed_tokens","lm_head"]

  use_wandb: true
  wandb_project: llava_linear
  wandb_name: llava_linear_pre

  num_workers: 8
  pin_memory: true
  prefetch_factor: 2
  rmpad: false
  rmpad_with_pos_ids: false

  empty_cache_steps: 500
  enable_forward_prefetch: false
  enable_gradient_checkpointing: true
  enable_mixed_precision: true
  enable_full_shard: true
  enable_reentrant: false

  ckpt_manager: dcp
  data_parallel_mode: fsdp1

  output_dir: ./output_dir/llava_linear_pre

**Comment by Luosuu** [comment]
hmm still I think we do not have enough information to know the reason. 

Please ensure you have proper _no_split_modules in your model like https://github.com/ByteDance-Seed/VeOmni/blob/e4e431d08076c6342c34fe3343d209fe385d8380/veomni/models/transformers/qwen3_moe/modeling_qwen3_moe.py#L573C5-L573C22

and maybe try fsdp2?

**Comment by Tttizi** [comment]
@Luosuu fsdp2在SFT的时候应该如何使用呢？

**Comment by Tttizi** [comment]
@Luosuu 使用fsdp2以后，grad_norm变成了0，似乎在模型初始化的时候会存在问题

**Comment by Tttizi** [comment]
@Luosuu 感谢您！！发现确实少定义了一个_no_split_modules，现在可以训练了。
